src/components/InputField.js

```diff
@@ -0,0 +1,20 @@
+import React from "react";
+
+export default class InputField extends React.Component {
+  constructor(props) {
+    super(props);
+    this.state = {
+      value: this.props.value
+    };
+  }
+
+  updateInputField(e) {
+    this.setState({value: e.nativeEvent.target.value});
+  }
+
+  render() {
+    return (
+        <input type="text" onBlur={() => this.props.callBack()} value={this.state.value} onChange={(e) => this.updateInputField(e)}/>
```

Comment:
Help me understand what onBlur and onChange do right here.  What I was hoping for is an "edit in place" behavior -- so no matter the user action that closes the "Edit" state of the text field, when it is closed, that update is sent to the backend.